iam.yml

```diff
@@ -157,6 +157,14 @@ Resources:
                   - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/apigateway/*"
                   - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/apigateway/"
                   - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/apigateway/*:log-stream:*"
+              - Effect: Allow
+                Action:
+                  - 'dynamodb:CreateTable'
+                Resource:
+                  - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*_blocked_users"
+                  - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*_tokens"
+                  - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*_user_requests"
+                  - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*_teacher_associated_requests"
```

Comment:
I'm curious about this specific filter pattern. Would something like `javabuilder_*` or `javabuilder_*_blocked_users" be more appropriate (with the `*` in the second case likely representing an environment name).

Reply:
Yeah that would be preferable -- the reason I left it more broad is that the way I'm setting up table names is by concatenating the subdomain name and `_blocked_users`, etc. We don't enforce that subdomain names need to start with javabuilder (although by convention, they have -- eg, javabuilderbeta, javabuilder-ben).

https://github.com/code-dot-org/javabuilder/pull/229

I didn't think that dynamoDB table names was the right place to enforce that javabuilder* prefix for subdomain names, but LMK if you disagree.

Reply:
That makes sense. Honestly a `*` wouldn't be inappropriate, as I think you could make a similar argument that this isn't the place to define what tables the application needs. This doesn't stop you from creating more than four tables, since you could easily create `pokemon_name_but_lets_call_it_user_requests`.

The cloudformation role needs the ability to create dynamo db tables, and unless we know the exact name, we're not so much restricting as we are enforcing a naming convention.

Reply:
Yeah good point -- would you prefer that I just wildcard it entirely so that if we ever add dynamodb tables in the future, we don't need to loop in infra?

Reply:
I think that would be fine. I will be revisiting this stack as a whole in the near future, and some of my thoughts for restructuring what gets deployed with which stack might open opportunities to more appropriate limit the access.